src/pool_worker_master.erl

```diff
@@ -0,0 +1,49 @@
+-module(pool_worker_master).
+% using list_to_integer as command line env variable declaration looks to set string
+-define(WORKER_NUMBER, list_to_integer(os:getenv("WORKER_NUMBER", "4"))).
+
+-export([init_pool_worker_master/0, find_worker_for_machine_execution/5]).
+
+find_worker_for_machine_execution(0, 3, _WorkerPidList, _MachineConfigToExecute, ReqActorPid) ->
+    ReqActorPid ! {error, could_not_find_available_actor};
+find_worker_for_machine_execution(
+    0, RetryCounter, WorkerPidList, MachineConfigToExecute, ReqActorPid
+) ->
+    receive
+    after 1000 ->
+        find_worker_for_machine_execution(
+            ?WORKER_NUMBER, RetryCounter + 1, WorkerPidList, MachineConfigToExecute, ReqActorPid
+        )
+    end;
```

Comment:
I think it would be more idiomatic to use `timer:sleep(1000)`. To me, we should reserve the use of `receive` for, actually, receiving signals.

[The official documentation](https://www.erlang.org/doc/reference_manual/expressions.html#receive) even talks about this construct, but it seems a bit a weird syntax to me to actually sleep for one second.

<img width="1029" alt="CleanShot 2022-06-17 at 19 00 55@2x" src="https://user-images.githubusercontent.com/29370468/174344498-d24a04f7-033f-470f-ad05-076a39be96f9.png">

Reply:
`timer:sleep(T)` even [uses this construct internally](https://github.com/erlang/otp/blob/a37a8ea81ceeee452daa933b2bbeca6c138a20b8/lib/stdlib/src/timer.erl#L214-L224).

Reply:
+1